cmd/query/app/server.go

```diff
@@ -297,48 +295,48 @@ func (s *Server) Start() error {
 
 	s.bgFinished.Add(1)
 	go func() {
-		s.logger.Info("Starting HTTP server", zap.Int("port", httpPort), zap.String("addr", s.queryOptions.HTTPHostPort))
+		s.telset.Logger.Info("Starting HTTP server", zap.Int("port", httpPort), zap.String("addr", s.queryOptions.HTTPHostPort))
 		var err error
 		if s.queryOptions.TLSHTTP.Enabled {
 			err = s.httpServer.ServeTLS(s.httpConn, "", "")
 		} else {
 			err = s.httpServer.Serve(s.httpConn)
 		}
 		if err != nil && !errors.Is(err, http.ErrServerClosed) && !errors.Is(err, cmux.ErrListenerClosed) && !errors.Is(err, cmux.ErrServerClosed) {
-			s.logger.Error("Could not start HTTP server", zap.Error(err))
+			s.telset.Logger.Error("Could not start HTTP server", zap.Error(err))
 		}
-		s.logger.Info("HTTP server stopped", zap.Int("port", httpPort), zap.String("addr", s.queryOptions.HTTPHostPort))
-		s.healthCheck.Set(healthcheck.Unavailable)
+		s.telset.Logger.Info("HTTP server stopped", zap.Int("port", httpPort), zap.String("addr", s.queryOptions.HTTPHostPort))
+		s.telset.ReportStatus(component.NewStatusEvent(component.StatusStopping))
```

Comment:
not Stopped?

Reply:
Also, I am not quite sure about the semantics of ReportStatus calls - whose status are we reporting? Since we only get a single callback function I assume it is meant to represent the status of the overall component (such as jaeger_query extension). We, however, are calling this callback multiple times, once per each server, which seems wrong. It's worth checking how the official OTLP Receiver does that, since it also starts and stops two servers (http and grpc).

Reply:
As in the switch case, it was set as healthcheck.Unavailable before so changed it to .StatusStopping to make it remain hc.Unavailable itslef, if set to Stopped it will be reported as hc.Broken

Reply:
> Also, I am not quite sure about the semantics of ReportStatus calls - whose status are we reporting? Since we only get a single callback function I assume it is meant to represent the status of the overall component (such as jaeger_query extension). We, however, are calling this callback multiple times, once per each server, which seems wrong. It's worth checking how the official OTLP Receiver does that, since it also starts and stops two servers (http and grpc).

Ok will look into the OTLP code and report back to you.

Reply:
PTAL, in the receiver function it is running a separate go routine and waits for an error, if an error is created it reports back the error.
In the Server code, calling ReportStatus for each server is definitely redundant. So, the Server Start() and Stop() functions can ReportStatus only once.